library/src/main/java/com/datatorrent/lib/io/fs/AbstractFileInputOperator.java

```diff
@@ -124,6 +124,34 @@
   protected transient long skipCount = 0;
   private transient OperatorContext context;
 
+  /**
+   * The time between sending batch tuples. Default value is 1 hour. A batch
+   * interval could consist of multiple scan intervals
+   */
+  private long batchIntervalMillis = 60 * 60 * 1000; //1 hour
```

Comment:
Hourly batch is just an example, you can have even smaller intervals as batches.
Regarding the abstraction for a batch tuple, what do you suggest? We have predefined types of batch control tuples. For a start, we wanted to have file control tuples - emitted when a file starts and ends - which is a useful event for a downstream operator.
With regards to batch specification here, there is just a single use case for scanning every n time units and reading whatever is present as a batch.